src/figuresClasses.ts

```diff
@@ -1,11 +1,78 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: Shape = 'triangle',
```

Comment:
The constructor parameters are being manually assigned to class properties. In TypeScript, if you declare a constructor parameter with an access modifier (public in this case), TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property. You can remove the manual assignments and leave the constructor body empty.